.eslintrc.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+module.exports = {
+  parser: 'babel-eslint',
+  plugins: [
+    'react',
+    'react-hooks',
+    'flowtype',
+    'jest'
+  ],
+  extends: [
+    'standard',
+    'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'plugin:flowtype/recommended'
+  ],
+  env: {
+    es6: true,
+    node: true,
+    browser: true,
+    'jest/globals': true
+  },
+  settings: {
+    flowtype: {
+      onlyFilesWithFlowAnnotation: true
+    }
+  },
+  ignorePatterns: [
+    '**/flow-typed/',
+    '**/reports/',
+    '**/node_modules/',
+    '**/ios/main.jsbundle',
+    '**/stylelint.config.js',
+    '**/dist/',
+    '**/lib-dist/'
+  ],
+  rules: {
+    'arrow-parens': ['error', 'as-needed'],
+    curly: ['error', 'all'],
+    'jsx-quotes': ['error', 'prefer-single'],
+    'max-len': ['warn', { code: 120 }],
+    'no-loop-func': 'error',
+    'prefer-const': 'error',
+    'prefer-object-spread': 'error',
+    'prefer-template': 'error',
+
+    'react/display-name': 'off',
+    'react/no-access-state-in-setstate': 'error',
+    'react/no-did-mount-set-state': 'warn',
+    'react/no-did-update-set-state': 'warn',
+    'react/no-unescaped-entities': 'off',
+    'react/no-unused-prop-types': 'warn',
+    'react/no-typos': 'error',
+    'react/prefer-es6-class': ['error', 'always'],
+
+    'flowtype/delimiter-dangle': ['error', 'never'],
+    'flowtype/newline-after-flow-annotation': ['error', 'always'],
+    'flowtype/no-primitive-constructor-types': 'error',
+    'flowtype/no-weak-types': 'warn',
+    'flowtype/object-type-delimiter': ['error', 'comma'],
+    'flowtype/require-exact-type': ['warn', 'always'],
+    'flowtype/require-parameter-type': ['error', { excludeArrowFunctions: true }],
+    'flowtype/require-return-type': ['error', 'always', {
+      excludeArrowFunctions: true,
+      annotateUndefined: 'never',
+      excludeMatching: ['^render$']
+    }],
+    'flowtype/semi': ['error', 'never'],
+    'flowtype/type-id-match': ['error', '^(iOS)?([A-Z][a-z0-9]+)+Type$'],
+    'flowtype/valid-syntax': 'warn',
+
+    'jest/consistent-test-it': 'error',
+    'jest/no-disabled-tests': 'error',
+    'jest/no-focused-tests': 'error',
+    'jest/no-identical-title': 'error',
+    'jest/no-test-prefixes': 'error',
+    'jest/prefer-to-be-null': 'error',
+    'jest/prefer-to-have-length': 'error',
+    'jest/valid-describe': 'error',
+    'jest/valid-expect': 'error'
+  },
+  overrides: [{
+    files: ['**/src/**/*.js'],
```

Comment:
If you think there is a better file matcher, let me know.

Reply:
why do we want this rule only for specific files?

Reply:
Otherwise I had to type all config files etc. which would lead to lots of errors and necessary fixes. Also we don't have flow in the root project at all, so it would not make sense at all to type these files.

Reply:
Thanks for explaining. I think the matcher should be sufficient in that case. 👍🏼

Reply:
But some config files already benefit from linting. That way we ignore linting. Previously we had separate configs for linting and source.

Reply:
This only disables one line which is missing flow annotation. Every other rule is enforced on all files.

Reply:
oh sorry did not get the indentation :)